app/scripts/directives/metrics.js

```diff
@@ -303,21 +309,55 @@ angular.module('openshiftConsole')
           }
         }
 
-        function update() {
-          var pod = scope.pod,
-              container = scope.options.selectedContainer,
-              start = Date.now() - scope.options.timeRange.value * 60 * 1000;
+        function getConfig(metric, id) {
+          if (scope.pod) {
+            return {
+              pod: scope.pod,
+              // some metrics (network, disk) are not available at container
+              // level (only at pod and node level)
+              containerName: metric.containerMetric ? scope.options.selectedContainer.name : "pod",
+              namespace: scope.pod.metadata.namespace,
+              metric: id
+            };
+          }
+
+          if (scope.deployment) {
+            return {
+              deployment: scope.deployment.metadata.name,
+              namespace: scope.deployment.metadata.namespace,
+              metric: id
+            };
+          }
+
+          return null;
+        }
+
+        // Make sure there are no errors or missing data before updating.
+        function canUpdate() {
+          if (scope.metricsError) {
+            return false;
+          }
+
+          if (scope.deployment) {
+            return true;
+          }
 
-          if (!pod || !container || scope.metricsError) {
+          return scope.pod && _.get(scope, 'options.selectedContainer');
+        }
+
+        function update() {
+          if (!canUpdate()) {
             return;
           }
 
-          // Leave the end time off to use the server's current time as the end
-          // time. This prevents an issue where the donut chart shows 0 for
-          // current usage if the client clock is ahead of the server clock.
           angular.forEach(scope.metrics, function(metric) {
             var promises = [];
 
+            // Leave the end time off to use the server's current time as the end
+            // time. This prevents an issue where the donut chart shows 0 for
+            // current usage if the client clock is ahead of the server clock.
+            var start = Date.now() - scope.options.timeRange.value * 60 * 1000;
```

Comment:
with this moved into the loop wont this now cause you to have potentially different start times for the different charts?